Libraries/VgDataFramework/vdfDataReader.cxx

```diff
@@ -105,7 +105,8 @@ bool vdfDataReader::hasData() const
 //-----------------------------------------------------------------------------
 bool vdfDataReader::failed() const
 {
-  return this->source()->status() == vdfDataSource::Invalid;
+  auto* const s = this->source();
+  return (s && (s->status() == vdfDataSource::Invalid));
```

Comment:
So, if source isn't valid, this will return `false` - is that intended?  Should it be

`return (!s || (s->status() == vdfDataSource::Invalid));`

Reply:
Hmm... It would be more correct to say that if you don't have a source, the answer that `failed()` gives you is nonsense; don't do that. That said, it looks like it never actually matters, but if it *did*, the objective would be that failure to set a source (which is how this would matter) should *not* be considered failure to read data.

I'm not explaining it very well, but anyway the upshot is I'm more inclined to leave it alone.